conf/distro/openxt-main.conf

```diff
@@ -16,44 +16,65 @@
+
+XENCLIENT_BUILD ?= "${@time.strftime('%y%m%d',time.gmtime())}"
+XENCLIENT_BUILD_DATE ?= "${@time.strftime('%y%m%d',time.gmtime())}"
+XENCLIENT_BUILD_BRANCH ?= "master"
+XENCLIENT_VERSION ?= "7.0.0"
+XENCLIENT_RELEASE ?= "7.0.0"
+XENCLIENT_TOOLS ?= "17.0.0.0"
+
+DISTRO_FEATURES = "alsa ext2 largefile usbhost wifi xattr pci x11 ipv4 ipv6 ${DISTRO_FEATURES_LIBC} multiarch pam"
 
 # Down the road this should probably be a machine config thing so it is possible
 # to have images that do not include selinux
 DISTRO_FEATURES_append += "selinux"
 
-PREFERRED_VERSION_linux-libc-headers ?= "${@"${PREFERRED_VERSION_linux-openxt}".rstrip('%')}"
-PREFERRED_VERSION_linux-libc-headers-nativesdk ?= "${@"${PREFERRED_VERSION_linux-openxt}".rstrip('%')}"
+# Build source packages if XENCLIENT_BUILD_SRC_PACKAGES is set to 1.
+INHERIT += "xenclient-src-package"
+
+# Collect info about sources used in build if XENCLIENT_COLLECT_SRC_INFO is
+# set to 1.
+INHERIT += "xenclient-src-info"
 
-PREFERRED_VERSION_dojosdk-native ?= "1.7.2"
-# ${MACHINE}.conf is overriden by ${DISTRO}.conf
-# xenmgr_data and sync-wui (uivm and syncui machines) apparently depends on
-# different versions, so preserve that.
-PREFERRED_VERSION_dojosdk-native_xenclient-syncui ?= "1.8.1"
+# OpenXT specific OE tasks
+INHERIT += "xenclient-customtask"
+INHERIT += "openxt-noempty-src"
+INHERIT += "openxt-images"
 
-# Force specific package versions
-PREFERRED_VERSION_grub = "2.00+AUTOINC%"
-PREFERRED_VERSION_lvm2 = "2.02.85"
-PREFERRED_VERSION_networkmanager = "0.9.2.0"
-PREFERRED_VERSION_network-manager-applet = "0.9.2.0"
-PREFERRED_VERSION_wpa-supplicant = "0.7.3"
+PACKAGE_CLASSES ?= "package_ipk"
 
-PREFERRED_VERSION_webkit-gtk = "1.8.3"
+STAGING_IDLDIR = "${STAGING_DATADIR}/idl"
 
-PREFERRED_VERSION_fontsproto = "2.1.3"
-PREFERRED_VERSION_gnome-keyring = "2.32.1"
+VIRTUAL-RUNTIME_keymaps = "xenclient-console-keymaps"
 
-PREFERRED_VERSION_refpolicy-mcs = "2.20141203"
+FILESYSTEM_PERMS_TABLES = "files/xc-fs-perms.txt"
+POLKITAUTH = ""
 
-# Select our preferred providers
-PREFERRED_PROVIDER_sysvinit-inittab = "sysvinit"
-PREFERRED_PROVIDER_virtual/java-initial = "cacao-initial"
-PREFERRED_PROVIDER_virtual/java-initial-native = "cacao-initial-native"
-PREFERRED_PROVIDER_virtual/java-native = "jamvm-native"
-PREFERRED_PROVIDER_virtual/javac-native = "ecj-bootstrap-native"
+ENABLE_BINARY_LOCALE_GENERATION = "1"
+LOCALE_UTF8_ONLY = "1"
 
-PREFERRED_VERSION_rsyslog = "7%"
+# overwrite debian mirror for screen, as the debian version it's based on (lenny) is in oldstable now
+DEBIAN_MIRROR_pn-screen = "http://archive.debian.org/debian/pool"
+
+# make hacks
+PARALLEL_MAKE_pn-gnumail = "-j 1"
+PARALLEL_MAKE_pn-gnumail-native = "-j 1"
+PARALLEL_MAKE_pn-shared-mime-info = "-j 1"
+PARALLEL_MAKE_pn-shared-mime-info-native = "-j 1"
+
+LICENSE_FLAGS_WHITELIST = "commercial_broadcom-wifi"
+LICENSE_FLAGS_WHITELIST += "commercial_microcode-ctl"
+
+XENCLIENT_PACKAGE_FEED_URI="file:///storage/ipk"
```

Comment:
Sure, but could you clarify by what you are referring to when you say "builder". To clarify BB_ENV_EXTRAWHITE, bitbake erases all environment varialbes to ensure no taiting from host machine except for any variable names listed in BB_ENV_WHITE and BB_ENV_EXTRAWHITE. Convention is that in the envoking environment defines any variables it wants bitbake to inherit in the BB_ENV_EXTRAWHITE environment variable. See the shell script build/bb in openxt.git, which is where I would add XENCLIENT_PACKAGE_FEED_URI. Then the "builder" exports what it wants to have set in XENCLIENT_PACKAGE_FEED_URI and it will override what is set in openxt-main.conf